### PR TITLE
Spin up httpbin container via pytest fixture

### PR DESCRIPTION
**Pull request by jwodder**
Part of #261.

### CODE REVIEW

File: datalad_next/tests/fixtures.py [hunk 1]
```diff
@@ -295,3 +298,57 @@ def http_server_with_basicauth(tmp_path_factory, http_credential):
...
+            else:
+                raise RuntimeError("httpbin container did not start up in time")
+        yield {
+            "standard": ciurl if container_id is not None else hbsurl,
+            "http": hburl,
+            "https": hbsurl,
+        }
+    finally:
+        if container_id is not None:
+            subprocess.run(["docker", "rm", "-f", container_id], check=True)
```
Comment (yarikoptic) [inline datalad_next/tests/fixtures.py:354]: May be just add `--rm` to `docker run` or there is benefit?

Comment (yarikoptic) [reply datalad_next/tests/fixtures.py:354]: Nevermind, it is a process in background, right

**Review by mih — APPROVED** [review]
Thank you! Maintains behavior, improves implementation. Much appreciated!

Note to self: Continues to ignore the possibility to use docker on windows. Watch failure pattern with public instance now and add later, if necessary.

**Comment by mih** [comment]
Macos test on appveyor now need protection against missing docker again.

**Review by mih — APPROVED** [review]

**Pull request merged**